### PR TITLE
Add selinuxd's selinux profile

### PR DESCRIPTION
**Pull request by JAORMX**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This adds a SELinux profile for selinuxd. These remove the need for
selinuxd to run as a privileged workload, and locks down the SPOD.

#### Which issue(s) this PR fixes:

None

#### Does this PR have test?

N/A.

#### Does this PR introduce a user-facing change?

```release-note
The workload that handles SELinux policy installation (selinuxd) is no longer a privileged container.
```

### CODE REVIEW

File: deploy/base/profiles/selinuxd.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
+{
+    "defaultAction": "SCMP_ACT_ERRNO",
+    "architectures": ["SCMP_ARCH_X86_64", "SCMP_ARCH_X86", "SCMP_ARCH_X32"],
+    "syscalls": [
+        {
+            "action": "SCMP_ACT_ALLOW",
+            "args": [],
+            "comment": "",
+            "excludes": {},
+            "includes": {},
```
Comment (saschagrunert) [inline deploy/base/profiles/selinuxd.json:10]: Let's remove those optional fields.

File: internal/pkg/manager/spod/bindata/spod.go [hunk 2]
```diff
@@ -131,12 +132,19 @@ var Manifest = &appsv1.DaemonSet{
...
+						semodule -i /etc/selinux.d/*.cil
 					`},
 						VolumeMounts: []v1.VolumeMount{
 							{
 								Name:      "selinux-drop-dir",
 								MountPath: SelinuxDropDirectory,
 							},
+							{
+								Name:      "operator-seccomp-profile-volume",
+								MountPath: "/opt/seccomp-profiles",
```
Comment (saschagrunert) [inline internal/pkg/manager/spod/bindata/spod.go:145]: We also add selinux policies into this directory, so I guess `seccomp-profiles` is misleading.

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
I think this is also worth a release note :)

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by JAORMX** [comment]
Seems the seccomp profile I generated is too restrictive. I'll only add selinux confinement for now and try to get a better seccomp profile.

**Comment by JAORMX** [comment]
Why do you hate me selinux?

**Comment by JAORMX** [comment]
Finally!!!

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by JAORMX** [comment]
/retest
```
         --- FAIL: TestSuite/TestSecurityProfilesOperator/cluster-wide:_Seccomp:_Verify_base_profile_merge (28.35s) 
```
:confused:

**Comment by JAORMX** [comment]
```
         --- FAIL: TestSuite/TestSecurityProfilesOperator/namespaced:_Seccomp:_Verify_base_profile_merge (27.19s) 
```
/retest

**Pull request merged**